File: packages/ruby/lib/logseal/resources/schemas.rb
# frozen_string_literal: true

module LogSeal
  module Resources
    class Schemas
      def initialize(client)
        @client = client
      end

      def list
        @client.request(:get, "/v1/schemas")
      end

      def create(action:, description: nil, target_types: nil, metadata_schema: nil)
        body = { action: }
        body[:description] = description if description
        body[:target_types] = target_types if target_types
        body[:metadata_schema] = metadata_schema if metadata_schema
        @client.request(:post, "/v1/schemas", body:)
      end

      def get(id)
        @client.request(:get, "/v1/schemas/#{id}")
      end

      def update(id, **fields)
        @client.request(:patch, "/v1/schemas/#{id}", body: fields)
      end

      def delete(id)
        @client.request(:delete, "/v1/schemas/#{id}")
      end
    end
  end
end

File: packages/php/src/Resource/Schemas.php
<?php

declare(strict_types=1);

namespace LogSeal\Resource;

use LogSeal\LogSeal;

class Schemas
{
    public function __construct(private readonly LogSeal $client) {}

    public function list(): array
    {
        return $this->client->request('GET', '/v1/schemas');
    }

    public function create(string $action, ?string $description = null, ?array $targetTypes = null, ?array $metadataSchema = null): array
    {
        $body = ['action' => $action];
        if ($description !== null) $body['description'] = $description;
        if ($targetTypes !== null) $body['target_types'] = $targetTypes;
        if ($metadataSchema !== null) $body['metadata_schema'] = $metadataSchema;
        return $this->client->request('POST', '/v1/schemas', $body);
    }

    public function get(string $id): array
    {
        return $this->client->request('GET', "/v1/schemas/{$id}");
    }

    public function update(string $id, array $fields): array
    {
        return $this->client->request('PATCH', "/v1/schemas/{$id}", $fields);
    }

    public function delete(string $id): void
    {
        $this->client->request('DELETE', "/v1/schemas/{$id}");
    }
}
